miden/tests/integration/stdlib/math/secp256k1_scalar_field.rs

```diff
@@ -0,0 +1,342 @@
+        pc = d[8];
+        c[3..11].copy_from_slice(&d[0..8]);
+
+        Self::u256xu32(&mut c[3..12], rhs.limbs[3], &self.limbs);
+
+        let d = Self::u288_reduce(&c[3..12], pc);
+        pc = d[8];
+        c[4..12].copy_from_slice(&d[0..8]);
+
+        Self::u256xu32(&mut c[4..13], rhs.limbs[4], &self.limbs);
+
+        let d = Self::u288_reduce(&c[4..13], pc);
+        pc = d[8];
+        c[5..13].copy_from_slice(&d[0..8]);
+
+        Self::u256xu32(&mut c[5..14], rhs.limbs[5], &self.limbs);
+
+        let d = Self::u288_reduce(&c[5..14], pc);
+        pc = d[8];
+        c[6..14].copy_from_slice(&d[0..8]);
+
+        Self::u256xu32(&mut c[6..15], rhs.limbs[6], &self.limbs);
+
+        let d = Self::u288_reduce(&c[6..15], pc);
+        pc = d[8];
+        c[7..15].copy_from_slice(&d[0..8]);
+
+        Self::u256xu32(&mut c[7..16], rhs.limbs[7], &self.limbs);
+
+        let d = Self::u288_reduce(&c[7..16], pc);
+        pc = d[8];
+        c[8..16].copy_from_slice(&d[0..8]);
+
+        let mut one = Self::one().limbs;
+        for i in 0..8 {
+            one[i] = one[i] * pc;
+        }
+
+        pc = 0;
+
+        (pc, c[8]) = Self::adc(c[8], one[0], pc);
+        (pc, c[9]) = Self::adc(c[9], one[1], pc);
+        (pc, c[10]) = Self::adc(c[10], one[2], pc);
+        (pc, c[11]) = Self::adc(c[11], one[3], pc);
+        (pc, c[12]) = Self::adc(c[12], one[4], pc);
+        (pc, c[13]) = Self::adc(c[13], one[5], pc);
+        (pc, c[14]) = Self::adc(c[14], one[6], pc);
+        (_, c[15]) = Self::adc(c[15], one[7], pc);
+
+        Self::Output {
+            limbs: c[8..16].try_into().expect("incorrect length"),
+        }
+    }
+}
+
+impl PartialEq for ScalarField {
+    /// Checks whether two secp256k1 scalarfield elements are equal or not, in Montogomery form
+    fn eq(&self, other: &Self) -> bool {
+        let mut flg = false;
+
+        for i in 0..8 {
+            flg |= (self.limbs[i] ^ other.limbs[i]) != 0;
+        }
+
+        !flg
+    }
+
+    /// Checks whether two secp256k1 scalar field elements are not equal to each other, in Montogomery form
+    fn ne(&self, other: &Self) -> bool {
+        !(self == other)
+    }
+}
+
+#[test]
+fn test_mul() {
+    let source = "
+    use.std::math::secp256k1_scalar_field
+
+    begin
+        exec.secp256k1_scalar_field::mul
+    end";
+
+    let mut stack = [0u64; 16];
+
+    let mut elm0 = ScalarField::zero();
+    let mut elm1 = ScalarField::zero();
+
+    for i in 0..8 {
+        let a = rand_utils::rand_value::<u32>();
+        let b = rand_utils::rand_value::<u32>();
+
+        stack[i] = a as u64;
+        stack[i ^ 8] = b as u64;
+
+        elm0.limbs[i] = a;
+        elm1.limbs[i] = b;
+    }
+
+    let elm2 = elm0 * elm1;
+    stack.reverse();
+
+    let test = build_test!(source, &stack);
+    let strace = test.get_last_stack_state();
+
+    for i in 0..8 {
+        assert_eq!(Felt::new(elm2.limbs[i] as u64), strace[i]);
+    }
+}
```

Comment:
Note for the future:

It looks like this test code is exactly the same as the code in the base field tests, with the exception of the `source` string and the initialization of `elm0` and `elm1`.

In cases like this, I'd prefer to encapsulate repeated code in separate helper functions. For example:
- in the math tests, create a new "parent" secp256k1 module and move all of the secp256k1 modules into it as submodules. 
- At the top level of the new module, define generic test helpers that accept the `source` string and the required inputs (e.g. `elm0`, `elm1`)
- In each of the secp256k1, call the test helper function, passing it the source string and the initialization values.

This will reduce duplication and should make our test code more robust.

Can you make a note about refactoring these tests in the issue about switching to using a library for the `BaseField` and `ScalarField` implementations? This will be easier to do after or during that change

Reply:
We'll move all these cryptographic logics from here and  modularize them in crypto crate. See https://github.com/0xPolygonMiden/crypto/issues/19 where I note it down.

Reply:
Looks good, thanks!